include/mqtt/endpoint.hpp

```diff
@@ -4216,41 +4217,7 @@ class endpoint : public std::enable_shared_from_this<endpoint<Mutex, LockGuard,
      */
     optional<packet_id_t> acquire_unique_packet_id_no_except() {
         LockGuard<Mutex> lck (store_mtx_);
-        if (packet_id_.size() == std::numeric_limits<packet_id_t>::max()) return nullopt;
-        if (packet_id_master_ == std::numeric_limits<packet_id_t>::max()) {
-            packet_id_master_ = 1U;
-        }
-        else {
-            ++packet_id_master_;
-        }
-        auto ret = packet_id_.insert(packet_id_master_);
-        if (ret.second) return packet_id_master_;
-
-        auto last = packet_id_.end();
-        auto e = last;
-        --last;
-
-        if (*last != std::numeric_limits<packet_id_t>::max()) {
-            packet_id_master_ = static_cast<packet_id_t>(*last + 1U);
-            packet_id_.insert(e, packet_id_master_);
-            return packet_id_master_;
-        }
-
-        auto b = packet_id_.begin();
-        auto prev = *b;
-        if (prev != 1U) {
-            packet_id_master_ = 1U;
-            packet_id_.insert(b, packet_id_master_);
-            return packet_id_master_;
-        }
-        ++b;
-        while (*b - 1U == prev && b != e) {
-            prev = *b;
-            ++b;
-        }
-        packet_id_master_ = static_cast<packet_id_t>(prev + 1U);
-        packet_id_.insert(b, packet_id_master_);
-        return packet_id_master_;
+        return pid_man_.acquire_unique_id();
```

Comment:
It seems that something wrong. I believe that the pid_man_ is already been merged to the master.
What is happening ?

Reply:
It is because I rebased incorrectly, let me correct this. 
If I do git rebase HEAD~4 --interactive, than this happens.